In [2]:
%load_ext sql
%sql postgresql+psycopg2://dataanalysis:dataanalysis@localhost:5432/project

'Connected: dataanalysis@project'

# Superficial analysis

## Name

As we saw, there are 5 people called the same. We can assume that, since Brandon Jones, Eric Harris and Michael Johnson
are pretty common names, they are different people...and maybe our friends Jamake and Daquan are some kind of mistake

In [4]:
%%sql 

select name, count(name) as total
        from victims
            group by name
            order by total desc
            limit 10

10 rows affected.


[('Brandon Jones', 2),
 ('Eric Harris', 2),
 ('Jamake Cason Thomas', 2),
 ('Michael Johnson', 2),
 ('Daquan Antonio Westbrook', 2),
 ('Michael T. Stoudt', 1),
 ('Dennis Claude Stanley', 1),
 ('Marvin Cremeans', 1),
 ('Martin Gomez', 1),
 ('Millard Tallant III', 1)]

but maybe it's better to be completely sure...

In [5]:
%%sql

select * from victims
    where name in ('Brandon Jones','Eric Harris','Jamake Cason Thomas','Michael Johnson','Daquan Antonio Westbrook')
    order by name

10 rows affected.


[(1991, 'Brandon Jones', datetime.date(2016, 10, 25), 'shot', 'toy weapon', 25, 'M', 'W', 'Elkton', 'MD', 'True', 'attack', 'Not fleeing', 'False'),
 (229, 'Brandon Jones', datetime.date(2015, 3, 19), 'shot', 'unarmed', 18, 'M', 'B', 'Cleveland', 'OH', 'False', 'attack', 'Not fleeing', 'False'),
 (1129, 'Daquan Antonio Westbrook', datetime.date(2015, 12, 24), 'shot', 'gun', 18, 'M', 'B', 'Charlotte', 'NC', 'False', 'attack', 'Not fleeing', 'False'),
 (2158, 'Daquan Antonio Westbrook', datetime.date(2016, 12, 24), 'shot', 'gun', 18, 'M', 'B', 'Charlotte', 'NC', 'False', 'undetermined', 'Not fleeing', 'False'),
 (291, 'Eric Harris', datetime.date(2015, 4, 2), 'shot', 'unarmed', 44, 'M', 'B', 'Tulsa', 'OK', 'False', 'other', 'Not fleeing', 'False'),
 (1252, 'Eric Harris', datetime.date(2016, 2, 8), 'shot', 'vehicle', 22, 'M', 'B', 'New Orleans', 'LA', 'False', 'other', 'Car', 'False'),
 (2294, 'Jamake Cason Thomas', datetime.date(2017, 2, 3), 'shot', 'gun', 22, 'M', 'B', 'Rowland', 'NC', 'False', 'attack', 'Car', 'False'),
 (2304, 'Jamake Cason Thomas', datetime.date(2017, 2, 3), 'shot', 'gun', 22, 'M', 'B', 'Rowland', 'NC', 'False', 'attack', 'Car', 'False'),
 (1526, 'Michael Johnson', datetime.date(2016, 5, 9), 'shot', 'gun', 26, 'M', 'W', 'Chicago', 'IL', 'False', 'other', 'Foot', 'False'),
 (1586, 'Michael Johnson', datetime.date(2016, 6, 1), 'shot', 'gun', 21, 'M', 'B', 'Newnan', 'GA', 'False', 'attack', 'Foot', 'True')]

Ok, comparing the data, we can say the Brandons, Erics and Michaels are different people, but Daquan and Jamake are the same.

The case of Jamake is easy: we just have to get rid of one of them, since the 2 entries are completely the same (except id).

The case of Daquan is different: 2 discrepancies between the entries. Date:2015 or 2016? Threat level:attack or undetermined?.

Googling daquan, we find he was killed dec. 2015, and that he pointed an officer with his gun...so, the right one is 1129

We can get rid of 2158

## Name null


In [6]:
%%sql

select * from victims
    where name is null
    order by date

19 rows affected.


[(1074, None, datetime.date(2015, 12, 10), 'shot', 'gun', None, 'M', None, 'Hemet', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (1570, None, datetime.date(2016, 5, 26), 'shot', 'gun', None, 'M', None, 'San Antonio', 'TX', 'False', 'other', 'Not fleeing', 'False'),
 (1581, None, datetime.date(2016, 5, 26), 'shot', 'knife', None, 'F', None, 'Sneads', 'FL', 'False', 'other', 'Not fleeing', 'False'),
 (1584, None, datetime.date(2016, 6, 1), 'shot', 'gun', None, 'M', None, 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1615, None, datetime.date(2016, 6, 9), 'shot', 'baton', None, 'M', None, 'Somerton', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1685, None, datetime.date(2016, 7, 4), 'shot', 'knife', None, 'M', None, 'Rosser', 'TX', 'False', 'other', 'Not fleeing', 'True'),
 (1848, None, datetime.date(2016, 9, 1), 'shot and Tasered', 'pipe', None, 'M', 'H', 'Huntington Park', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (2037, None, datetime.date(2016, 11, 11), 'shot and Tasered', None, None, 'M', None, 'East Point', 'GA', 'False', 'undetermined', None, 'False'),
 (2110, None, datetime.date(2016, 12, 8), 'shot', 'gun', None, 'M', None, 'Allen', 'TX', 'False', 'attack', 'Not fleeing', 'False'),
 (2141, None, datetime.date(2016, 12, 20), 'shot', 'knife', None, 'M', None, 'Brawley', 'CA', 'True', 'other', 'Not fleeing', 'False'),
 (2154, None, datetime.date(2016, 12, 21), 'shot', 'knife', None, 'M', None, 'Stockton', 'CA', 'False', 'undetermined', 'Not fleeing', 'False'),
 (2168, None, datetime.date(2016, 12, 23), 'shot', 'vehicle', None, 'M', None, 'El Monte', 'CA', 'False', 'other', 'Car', 'False'),
 (2164, None, datetime.date(2016, 12, 24), 'shot', 'gun', None, 'M', None, 'Gadsden', 'AL', 'False', 'attack', 'Foot', 'False'),
 (2182, None, datetime.date(2016, 12, 30), 'shot', 'gun', None, 'M', None, 'Pensacola', 'FL', 'False', 'attack', 'Not fleeing', 'False'),
 (2232, None, datetime.date(2017, 1, 18), 'shot', 'hatchet', None, 'M', None, 'Los Angeles', 'CA', 'False', 'other', 'Car', 'False'),
 (2278, None, datetime.date(2017, 1, 31), 'shot and Tasered', 'knife', None, 'M', 'B', 'Hollywood', 'CA', 'True', 'attack', 'Not fleeing', 'False'),
 (2319, None, datetime.date(2017, 2, 9), 'shot', 'undetermined', None, 'M', None, 'Crownpoint', 'NM', 'False', 'undetermined', None, 'False'),
 (2328, None, datetime.date(2017, 2, 12), 'shot', 'vehicle', None, 'M', None, 'Terrell', 'TX', 'False', 'other', 'Other', 'False'),
 (2384, None, datetime.date(2017, 2, 25), 'shot', 'gun', None, None, None, 'Lumpkin', 'GA', 'False', 'attack', 'Car', 'False')]

This is our list of John Does...noone of them has age neither,most of them no race...but, we have 4 data to play to know what to do with them: date,armed,gender and city...it´s difficult that 2 people were killed the same day, with the same weapon, same gend)er in the
same place. We have to check which are the most reliable columns.

For the time being, and for future research, our johns and janes are:

    1074,1570,1581,1584,1615,1685,1848,2037,2110,2141,2154,2168,2164,2182,2232,2278,2319,2328,2384

In [8]:
%%sql

select
    count(id) as total,
    count(date) as dates,
    count(gender) as genders,
    count (city) as cities,
    count(armed) as weapons
from victims;


1 rows affected.


[(2142, 2142, 2141, 2142, 2136)]

Every entry has date and city, and we have just one gender missing..and he is someone without a name to (has we can see if we check the null names list. This case we can treat it apart...we can extract from here that, to check if this null names entries are an error, we can rely on date, city and gender, but we have 6 missing weapons...let´s try with three first

In [10]:
%%sql

select name,date,gender,city
from victims

where (date = '2015-12-10' and gender = 'M' and city = 'Hemet')
or
(date = '2016-05-26' and gender = 'M' and city = 'San Antonio')
or
(date = '2016-05-26' and gender = 'F' and city = 'Sneads')
or
(date = '2016-06-01' and gender = 'M' and city = 'Phoenix')
or
(date = '2016-06-09' and gender = 'M' and city = 'Somerton')
or
(date = '2016-07-04' and gender = 'M' and city = 'Rosser')
or
(date = '2016-09-01' and gender = 'M' and city = 'Huntington Park')
or
(date = '2016-11-11' and gender = 'M' and city = 'East Point')
or
(date = '2016-12-08' and gender = 'M' and city = 'Allen')
or
(date = '2016-12-20' and gender = 'M' and city = 'Brawley')
or
(date = '2016-12-21' and gender = 'M' and city = 'Stockton')
or
(date = '2016-12-23' and gender = 'M' and city = 'El Monte')
or
(date = '2016-12-24' and gender = 'M' and city = 'Gadsden')
or
(date = '2016-12-30' and gender = 'M' and city = 'Pensacola')
or
(date = '2017-01-18' and gender = 'M' and city = 'Los Angeles')
or
(date = '2017-01-31' and gender = 'M' and city = 'Hollywood')
or
(date = '2017-02-09' and gender = 'M' and city = 'Crownpoint')
or
(date = '2017-02-12' and gender = 'M' and city = 'Terrell')
or
(date = '2017-02-25' and city = 'Lumpkin')

order by date


19 rows affected.


[(None, datetime.date(2015, 12, 10), 'M', 'Hemet'),
 (None, datetime.date(2016, 5, 26), 'M', 'San Antonio'),
 (None, datetime.date(2016, 5, 26), 'F', 'Sneads'),
 (None, datetime.date(2016, 6, 1), 'M', 'Phoenix'),
 (None, datetime.date(2016, 6, 9), 'M', 'Somerton'),
 (None, datetime.date(2016, 7, 4), 'M', 'Rosser'),
 (None, datetime.date(2016, 9, 1), 'M', 'Huntington Park'),
 (None, datetime.date(2016, 11, 11), 'M', 'East Point'),
 (None, datetime.date(2016, 12, 8), 'M', 'Allen'),
 (None, datetime.date(2016, 12, 20), 'M', 'Brawley'),
 (None, datetime.date(2016, 12, 21), 'M', 'Stockton'),
 (None, datetime.date(2016, 12, 23), 'M', 'El Monte'),
 (None, datetime.date(2016, 12, 24), 'M', 'Gadsden'),
 (None, datetime.date(2016, 12, 30), 'M', 'Pensacola'),
 (None, datetime.date(2017, 1, 18), 'M', 'Los Angeles'),
 (None, datetime.date(2017, 1, 31), 'M', 'Hollywood'),
 (None, datetime.date(2017, 2, 9), 'M', 'Crownpoint'),
 (None, datetime.date(2017, 2, 12), 'M', 'Terrell'),
 (None, datetime.date(2017, 2, 25), None, 'Lumpkin')]

from here we can see that there are no missing entries that are similar (as it happened with the repeated names)

Since we have 19 cases, we will have to go one by one. The only thing we can do is avoid all of this being a typo.
I mean, we will have to check dates and cities. The procedure is the same, changing data, so I can copy to go faster changing date and city as required. We will have to use the -- thing to go through each query

In [11]:
%%sql

-- 1: 1074, from Hemet,2015-12-10

select * from victims where city = 'Hemet' order by name desc;
select * from victims where date = '2015-12-10' order by name desc;


6 rows affected.
4 rows affected.


[(1074, None, datetime.date(2015, 12, 10), 'shot', 'gun', None, 'M', None, 'Hemet', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (1073, 'Thomas Arthur Gendreau', datetime.date(2015, 12, 10), 'shot', 'knife', 54, 'M', 'W', 'Marina', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (1071, 'Jason Bryant', datetime.date(2015, 12, 10), 'shot', 'gun', 34, 'M', 'W', 'Indianapolis', 'IN', 'False', 'attack', 'Not fleeing', 'False'),
 (1072, 'Charles Edward Rosemond', datetime.date(2015, 12, 10), 'shot', 'gun', 55, 'M', 'B', 'Taylors', 'SC', 'False', 'attack', 'Not fleeing', 'False')]

 In the city query, we have discovered Nick Hamilton, with race missing, and 'other' in attack and flee. The date can be a typo...
or not...2016-12-20 vs 2015-12-10
The data query shows us no possible typo in the city entry
Googling says us they are 2 different cases, no typo, and indeed 1074 is a case
As a resume, man on parole with a felony warrant, brandishing a handgun in a parking slot of a mall, name not released by the police
KilledByPolice sheds no light on this case
We will have to think in creating a John Doe

In [12]:
%%sql


-- 2: 1570, from San Antonio, 2016-05-26

select * from victims where city = 'San Antonio' order by name desc;
select * from victims where date = '2016-05-26' order by name desc;

16 rows affected.
7 rows affected.


[(1581, None, datetime.date(2016, 5, 26), 'shot', 'knife', None, 'F', None, 'Sneads', 'FL', 'False', 'other', 'Not fleeing', 'False'),
 (1570, None, datetime.date(2016, 5, 26), 'shot', 'gun', None, 'M', None, 'San Antonio', 'TX', 'False', 'other', 'Not fleeing', 'False'),
 (1569, 'Warren Christian', datetime.date(2016, 5, 26), 'shot', 'gun', 43, 'M', 'W', 'Lansing', 'IL', 'False', 'attack', 'Foot', 'False'),
 (1572, 'Fernando Castro', datetime.date(2016, 5, 26), 'shot', 'gun', 19, 'M', 'H', 'Solvang', 'CA', 'False', 'attack', 'Car', 'False'),
 (1568, 'Eugene "Geno" Francis Smith', datetime.date(2016, 5, 26), 'shot', 'gun', 29, 'M', 'W', 'St. Paul', 'MN', 'False', 'attack', 'Not fleeing', 'False'),
 (1571, 'Devonte Gates', datetime.date(2016, 5, 26), 'shot', 'undetermined', 21, 'M', 'B', 'Washington Park', 'IL', 'False', 'undetermined', 'Foot', 'False'),
 (1583, 'Angel Daniel Navarro', datetime.date(2016, 5, 26), 'shot', 'gun', 25, 'M', 'H', 'Socorro', 'NM', 'False', 'attack', 'Car', 'False')]

Here the city is messy, because is a big city. We have 16 cases. But the only possible typo in the dates is Joe Angel Cisneros,possible typo in the date 2015-06-26 vs 2016-05-26...but the flee are different

In the date query we have two things: no typo possible, and the 1581 case (our next). Just to clarify, they are not the same different weapon,gender,city.

Googling, though, gives us interesting info. Angel Daniel Navarro, killed for hijacking with a gun, n the Interstate 25, between San Antonio and Socorro...again, we will have to drop this loose end, since, it's another S.Antonio, not the one in Texas, and the Socorro guy was fleeing by car. We wil have to live with the fact that the Sanantonio.gov web says something about the incident...but the page is not working.

Killed by police shares two things: the man was 45,latin, and according to the news, the police stated he was shot and tased, not just shot

So, we have another John Doe, but we can add some info


In [13]:
%%sql

-- 3: 1581, from Sneads, 2016-05-26

select * from victims where city = 'Sneads' order by name desc;

1 rows affected.


[(1581, None, datetime.date(2016, 5, 26), 'shot', 'knife', None, 'F', None, 'Sneads', 'FL', 'False', 'other', 'Not fleeing', 'False')]


No typo in the city, has we said in the previous case, so we can spare this query. The city query offers this has been the only victim in Sneads.

Googling says the case indeed happened, no name released. Woman try to attack in a case of domestic violence with a knife
in the presence of an officer.

According to Killed by police, she was Tonnia Reshelle Jarvis, 48, white


In [24]:
%%sql

-- 4: 1584, from Phoenix,2016-06-01

select * from victims where city = 'Phoenix' order by name desc;

24 rows affected.


[(1584, None, datetime.date(2016, 6, 1), 'shot', 'gun', None, 'M', None, 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1314, 'Victor Rivera', datetime.date(2016, 2, 24), 'shot', 'vehicle', 27, 'M', 'H', 'Phoenix', 'AZ', 'False', 'other', 'Car', 'False'),
 (1864, 'Steven Del Rio', datetime.date(2016, 9, 6), 'shot', 'undetermined', 31, 'M', 'H', 'Phoenix', 'AZ', 'False', 'undetermined', 'Car', 'False'),
 (1290, 'Sherrisa Homer', datetime.date(2016, 2, 21), 'shot', 'gun', 36, 'F', 'N', 'Phoenix', 'AZ', 'True', 'attack', 'Foot', 'False'),
 (775, 'Shane Rudolph', datetime.date(2015, 8, 25), 'shot', 'gun', 45, 'M', 'W', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1781, 'Ruben Horacio Strand Alvear', datetime.date(2016, 8, 13), 'shot', 'knife', 38, 'M', 'H', 'Phoenix', 'AZ', 'True', 'attack', 'Not fleeing', 'False'),
 (623, 'Robert Hammonds', datetime.date(2015, 7, 9), 'shot', 'gun', 68, 'M', 'W', 'Phoenix', 'AZ', 'False', 'attack', 'Foot', 'False'),
 (1884, 'Robert Carrillo', datetime.date(2016, 9, 15), 'shot', 'gun', 32, 'M', 'H', 'Phoenix', 'AZ', 'False', 'other', 'Other', 'False'),
 (348, 'Nicolas Leland Tewa', datetime.date(2015, 1, 27), 'shot', 'vehicle', 26, 'M', 'N', 'Phoenix', 'AZ', 'False', 'other', 'Not fleeing', 'False'),
 (1121, 'Lonnie Niesen', datetime.date(2015, 12, 26), 'shot', 'brick', 41, 'M', 'W', 'Phoenix', 'AZ', 'True', 'other', 'Not fleeing', 'False'),
 (1186, 'Levi Gene Wilson', datetime.date(2016, 1, 18), 'shot', 'gun', 39, 'M', 'W', 'Phoenix', 'AZ', 'True', 'attack', 'Not fleeing', 'False'),
 (652, 'Kevin Thomas Snyder', datetime.date(2015, 7, 18), 'shot', 'gun', 46, 'M', 'W', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (301, 'Ken Cockerel', datetime.date(2015, 4, 5), 'shot', 'knife', 51, 'M', 'W', 'Phoenix', 'AZ', 'True', 'other', 'Not fleeing', 'False'),
 (2205, 'JR Williams', datetime.date(2017, 1, 9), 'shot', 'unarmed', 38, 'M', 'B', 'Phoenix', 'AZ', 'False', 'other', 'Foot', 'True'),
 (1800, 'Jorge Ceniceros', datetime.date(2016, 8, 19), 'shot', 'gun', 22, 'M', 'H', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1941, 'John Ethan Carpentier', datetime.date(2016, 9, 27), 'shot', 'vehicle', 26, 'M', 'W', 'Phoenix', 'AZ', 'False', 'other', 'Car', 'False'),
 (1546, 'Israel Santos-Banos', datetime.date(2016, 5, 18), 'shot', 'gun', 20, 'M', 'H', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1537, 'Francis Clark', datetime.date(2016, 5, 14), 'shot', 'gun', 43, 'M', 'N', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'True'),
 (376, 'Erik Tellez', datetime.date(2015, 4, 18), 'shot', 'gun', 43, 'M', 'H', 'Phoenix', 'AZ', 'True', 'attack', 'Not fleeing', 'False'),
 (1868, 'Dylan Papa', datetime.date(2016, 9, 6), 'shot', 'sword', 25, 'M', 'W', 'Phoenix', 'AZ', 'True', 'attack', 'Not fleeing', 'False'),
 (2186, 'Dustin Selby', datetime.date(2016, 12, 30), 'shot and Tasered', 'knife', 31, 'M', None, 'Phoenix', 'AZ', 'False', 'other', 'Foot', 'False'),
 (1820, 'Angel Torres Jr.', datetime.date(2016, 8, 27), 'shot', 'toy weapon', 26, 'M', 'H', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1317, 'Alex Buckner', datetime.date(2016, 2, 23), 'shot', 'gun', 26, 'M', 'W', 'Phoenix', 'AZ', 'True', 'attack', 'Not fleeing', 'False'),
 (271, 'Adam Reinhart', datetime.date(2015, 3, 7), 'shot', 'gun', 29, 'M', 'W', 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False')]

In [25]:
%sql select * from victims where date = '2016-06-01' order by name desc;

3 rows affected.


[(1584, None, datetime.date(2016, 6, 1), 'shot', 'gun', None, 'M', None, 'Phoenix', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1586, 'Michael Johnson', datetime.date(2016, 6, 1), 'shot', 'gun', 21, 'M', 'B', 'Newnan', 'GA', 'False', 'attack', 'Foot', 'True'),
 (1587, 'Daniel Pigg', datetime.date(2016, 6, 1), 'shot', 'gun', 51, 'M', 'W', 'Berea', 'KY', 'False', 'attack', 'Foot', 'False')]

Another tricky case in a very big city. 24 entries to try to see if there is a typo in the date: no typo

Date query: no typo in the city
Google says the name of the victim involved is Kevin T. Marshall, 30.
Killed by police adds white
Curiosly, the same day in phoenix a guy was struck and killed by a police car, 18-year-old Jordan Black (accident, not counted in these lists)


In [22]:
%%sql

-- 5: 1615, from Somerton, 2016-06-09

select * from victims where city = 'Somerton' order by name desc;

1 rows affected.


[(1615, None, datetime.date(2016, 6, 9), 'shot', 'baton', None, 'M', None, 'Somerton', 'AZ', 'False', 'attack', 'Not fleeing', 'False')]

In [26]:
%sql select * from victims where date = '2016-06-09' order by name desc;

5 rows affected.


[(1615, None, datetime.date(2016, 6, 9), 'shot', 'baton', None, 'M', None, 'Somerton', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (1603, 'William Michael Hollis', datetime.date(2016, 6, 9), 'shot', 'gun', 34, 'M', 'W', 'Birmingham', 'AL', 'False', 'attack', 'Car', 'False'),
 (1606, 'John Michael Brisco', datetime.date(2016, 6, 9), 'shot', 'gun', 52, 'M', 'B', 'Port Arthur', 'TX', 'False', 'other', 'Foot', 'False'),
 (1605, 'James Allen Nickol', datetime.date(2016, 6, 9), 'shot', 'gun', 38, 'M', 'W', 'York', 'PA', 'False', 'attack', None, 'False'),
 (1607, 'Clifford Tucker', datetime.date(2016, 6, 9), 'shot', 'gun', 68, 'M', 'W', 'Ishpeming Township', 'MI', 'True', 'attack', 'Not fleeing', 'True')]

In [27]:
%sql select * from victims where armed = 'baton' order by name desc;

2 rows affected.


[(1615, None, datetime.date(2016, 6, 9), 'shot', 'baton', None, 'M', None, 'Somerton', 'AZ', 'False', 'attack', 'Not fleeing', 'False'),
 (2062, 'Erickson Brito', datetime.date(2016, 11, 19), 'shot', 'baton', 21, 'M', 'B', 'Brooklyn', 'NY', 'True', 'attack', 'Not fleeing', 'False')]

In Somerton there has been not other killing but this one.
No typo on the city
Since the weapon is kind of particular, we do a 3rd query, but to typo there
Google says nothing. Killed by police conts on the 10 june an unkown guy,killed in Somerton, carrying a baton he stole from the agent...he was an unknown mexican citizen, he was latin


In [28]:
%%sql 

-- 6: 1685, from Rosser,2016-07-04

select * from victims where city = 'Rosser' order by name desc;


1 rows affected.


[(1685, None, datetime.date(2016, 7, 4), 'shot', 'knife', None, 'M', None, 'Rosser', 'TX', 'False', 'other', 'Not fleeing', 'True')]

In [19]:
%%sql

select * from victims where date = '2016-07-04' order by name desc;

4 rows affected.


[(1685, None, datetime.date(2016, 7, 4), 'shot', 'knife', None, 'M', None, 'Rosser', 'TX', 'False', 'other', 'Not fleeing', 'True'),
 (1679, 'William Tracy Patterson', datetime.date(2016, 7, 4), 'shot', 'gun', 34, 'M', 'W', 'North Charleston', 'SC', 'False', 'attack', 'Not fleeing', 'True'),
 (1683, 'Sidney Washington', datetime.date(2016, 7, 4), 'shot', 'gun', 21, 'M', 'B', 'Washington', 'DC', 'False', 'attack', 'Not fleeing', 'False'),
 (1681, 'Anthony Nunez', datetime.date(2016, 7, 4), 'shot', 'gun', 18, 'M', 'H', 'San Jose', 'CA', 'True', 'other', 'Not fleeing', 'False')]

Only one guy from Rosser
No typo on the city
Google says it happened, a knife wielding suspect killed after he stabbed 3 people
Killed by police says nothing more

In [29]:
%%sql

-- 7: 1848,from Huntington Park,2016-09-01

select * from victims where city = 'Huntington Park' order by name desc;

1 rows affected.


[(1848, None, datetime.date(2016, 9, 1), 'shot and Tasered', 'pipe', None, 'M', 'H', 'Huntington Park', 'CA', 'False', 'other', 'Not fleeing', 'False')]

In [31]:
%sql select * from victims where date = '2016-09-01' order by name desc;

4 rows affected.


[(1848, None, datetime.date(2016, 9, 1), 'shot and Tasered', 'pipe', None, 'M', 'H', 'Huntington Park', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (1847, 'Joshua Quintero', datetime.date(2016, 9, 1), 'shot', 'gun', 23, 'M', 'H', 'Florence', 'CA', 'False', 'other', 'Foot', 'False'),
 (1839, 'Ed Holup', datetime.date(2016, 9, 1), 'shot', 'vehicle', 60, 'M', 'W', 'Sylvania Township', 'OH', 'True', 'other', 'Not fleeing', 'False'),
 (1843, 'Caleb J. Douglas', datetime.date(2016, 9, 1), 'shot', 'gun', 18, 'M', 'W', 'Wichita', 'KS', 'False', 'other', 'Car', 'True')]

In [32]:
%sql select * from victims where armed = 'pipe' order by name desc;


1 rows affected.


[(1848, None, datetime.date(2016, 9, 1), 'shot and Tasered', 'pipe', None, 'M', 'H', 'Huntington Park', 'CA', 'False', 'other', 'Not fleeing', 'False')]

Only victim from Huntington Park

Not a typo in the city neither

Only guy armed with a pipe

Google says he was in his 35-40, when he was detained he was armed with rocks, and that he was previously tased

Killed by police says he was latin


In [33]:
%%sql

-- 8: 2037,from East Point,2016-11-11

select * from victims where city = 'East Point' order by name desc;

2 rows affected.


[(2037, None, datetime.date(2016, 11, 11), 'shot and Tasered', None, None, 'M', None, 'East Point', 'GA', 'False', 'undetermined', None, 'False'),
 (1767, 'Jamarion Rashad Robinson', datetime.date(2016, 8, 5), 'shot', 'gun', 26, 'M', 'B', 'East Point', 'GA', 'True', 'attack', 'Not fleeing', 'False')]

In [34]:
%sql select * from victims where date = '2016-11-11' order by name desc;

4 rows affected.


[(2037, None, datetime.date(2016, 11, 11), 'shot and Tasered', None, None, 'M', None, 'East Point', 'GA', 'False', 'undetermined', None, 'False'),
 (2046, 'William Ray Score', datetime.date(2016, 11, 11), 'shot', 'gun', 47, 'M', None, 'Montrose County', 'CO', 'False', 'attack', 'Not fleeing', 'False'),
 (2043, 'Marco Romero', datetime.date(2016, 11, 11), 'shot', 'gun', None, 'M', 'H', 'Boise', 'ID', 'False', 'attack', 'Other', 'False'),
 (2039, 'Brian Gaither', datetime.date(2016, 11, 11), 'shot', 'vehicle', 24, 'M', 'W', 'Grand Junction', 'CO', 'False', 'attack', 'Car', 'False')]

No typo in the date

No typo in the city

Google says that he was unarmed, not fleeing

Killed by police says he was Andrew Depeiza, 30 , Black

In [35]:
%%sql

-- 9: 2110, from Allen,2016-12-08

select * from victims where city = 'Allen' order by name desc;

2 rows affected.


[(2110, None, datetime.date(2016, 12, 8), 'shot', 'gun', None, 'M', None, 'Allen', 'TX', 'False', 'attack', 'Not fleeing', 'False'),
 (2187, 'Mark Guirguis', datetime.date(2017, 1, 1), 'shot', 'vehicle', 21, 'M', 'B', 'Allen', 'TX', 'False', 'attack', 'Car', 'False')]

In [37]:
%%sql


select * from victims where date = '2016-12-08' order by name desc;



1 rows affected.


[(2110, None, datetime.date(2016, 12, 8), 'shot', 'gun', None, 'M', None, 'Allen', 'TX', 'False', 'attack', 'Not fleeing', 'False')]

No typo in the date

No typo in the city

Google says in his 60s

Killed by police says  in his 60s

In [38]:
%%sql

-- 10: 2141, from Brawley,2016-12-20

select * from victims where city = 'Brawley' order by name desc;

1 rows affected.


[(2141, None, datetime.date(2016, 12, 20), 'shot', 'knife', None, 'M', None, 'Brawley', 'CA', 'True', 'other', 'Not fleeing', 'False')]

In [41]:
%sql select * from victims where date = '2016-12-20' order by name desc;

3 rows affected.


[(2141, None, datetime.date(2016, 12, 20), 'shot', 'knife', None, 'M', None, 'Brawley', 'CA', 'True', 'other', 'Not fleeing', 'False'),
 (2139, 'Nick Hamilton', datetime.date(2016, 12, 20), 'shot', 'gun', 31, 'M', None, 'Hemet', 'CA', 'False', 'other', 'Other', 'False'),
 (2140, 'Martin Gomez', datetime.date(2016, 12, 20), 'shot', 'knife', 46, 'M', 'H', 'Pharr', 'TX', 'False', 'other', 'Not fleeing', 'False')]

No typo in the date

No typo in the city

Google says maybe the date is 19th

Killed by police says Walter Ricardo Pimentel , Latin


In [42]:
%%sql

-- 11: 2154, from Stockton,2016-12-21

select * from victims where city = 'Stockton' order by name desc;

7 rows affected.


[(2154, None, datetime.date(2016, 12, 21), 'shot', 'knife', None, 'M', None, 'Stockton', 'CA', 'False', 'undetermined', 'Not fleeing', 'False'),
 (1459, 'Rodney Watts', datetime.date(2016, 4, 13), 'shot', 'undetermined', 35, 'M', 'B', 'Stockton', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (19, 'Patrick Wetter', datetime.date(2015, 1, 6), 'shot and Tasered', 'knife', 25, 'M', 'W', 'Stockton', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (346, 'Matautu Nuu', datetime.date(2015, 1, 28), 'shot and Tasered', 'hammer', 35, 'M', 'A', 'Stockton', 'CA', 'True', 'attack', 'Not fleeing', 'False'),
 (547, 'Kenneth Garcia', datetime.date(2015, 6, 14), 'shot', 'vehicle', 28, 'M', 'H', 'Stockton', 'CA', 'False', 'attack', 'Car', 'False'),
 (1791, 'Colby Friday', datetime.date(2016, 8, 16), 'shot', 'gun', 30, 'M', 'B', 'Stockton', 'CA', 'False', 'other', 'Foot', 'True'),
 (249, 'Carl Lao', datetime.date(2015, 3, 4), 'shot', 'gun', 28, 'M', 'A', 'Stockton', 'CA', 'False', 'attack', 'Not fleeing', 'False')]

In [43]:
%sql select * from victims where date = '2016-12-21' order by name desc;

8 rows affected.


[(2154, None, datetime.date(2016, 12, 21), 'shot', 'knife', None, 'M', None, 'Stockton', 'CA', 'False', 'undetermined', 'Not fleeing', 'False'),
 (2151, 'William Newman', datetime.date(2016, 12, 21), 'shot', 'vehicle', 46, 'M', None, 'Knox', 'IN', 'False', 'other', None, 'False'),
 (2152, 'Raymond Salaiz', datetime.date(2016, 12, 21), 'shot', 'vehicle', 38, 'M', 'H', 'Reno', 'NV', 'False', 'attack', 'Car', 'False'),
 (2156, 'Matthew Blake Richardson', datetime.date(2016, 12, 21), 'shot', 'gun', 27, 'M', 'W', 'Saraland', 'AL', 'False', 'attack', 'Not fleeing', 'False'),
 (2146, 'Ian Shea King', datetime.date(2016, 12, 21), 'shot', 'tire iron', 31, 'M', None, 'Ponca City', 'OK', 'False', 'other', 'Not fleeing', 'False'),
 (2147, 'Colton Calaway', datetime.date(2016, 12, 21), 'shot', 'gun', 25, 'M', 'W', 'Burns Flat', 'OK', 'False', 'attack', 'Other', 'False'),
 (2153, 'Benjamin Ortiz', datetime.date(2016, 12, 21), 'shot', 'knife', 25, 'M', None, 'Fayetteville', 'AR', 'True', 'other', 'Not fleeing', 'False'),
 (2157, 'Anthony Lovell Eddington', datetime.date(2016, 12, 21), 'shot', 'gun', 24, 'M', 'B', 'Coalinga', 'CA', 'False', 'attack', 'Not fleeing', 'False')]

No typo in the date

No typo in the city

Google says Luis Ambrosio-Aguilar, 30, latin

Killed by police says


In [44]:
%%sql

-- 12: 2168, from El Monte,2016-12-23

select * from victims where city = 'El Monte' order by name desc;

2 rows affected.


[(2168, None, datetime.date(2016, 12, 23), 'shot', 'vehicle', None, 'M', None, 'El Monte', 'CA', 'False', 'other', 'Car', 'False'),
 (879, 'William Chau', datetime.date(2015, 9, 9), 'shot', 'knife', 59, 'M', 'A', 'El Monte', 'CA', 'False', 'attack', 'Not fleeing', 'False')]

In [45]:
%sql select * from victims where date = '2016-12-23' order by name desc;


3 rows affected.


[(2168, None, datetime.date(2016, 12, 23), 'shot', 'vehicle', None, 'M', None, 'El Monte', 'CA', 'False', 'other', 'Car', 'False'),
 (2167, 'Stanley Eversol', datetime.date(2016, 12, 23), 'shot', 'crossbow', 55, 'M', None, 'Palm Harbor', 'FL', 'False', 'attack', 'Not fleeing', 'False'),
 (2165, 'Michael Tyler Jacques', datetime.date(2016, 12, 23), 'shot and Tasered', 'undetermined', 31, 'M', 'W', 'Bend', 'OR', 'False', 'undetermined', 'Not fleeing', 'False')]

In [46]:
%sql select * from victims where armed = 'vehicle' and state = 'CA' and (extract(year from date)=2016) order by name desc;


12 rows affected.


[(2168, None, datetime.date(2016, 12, 23), 'shot', 'vehicle', None, 'M', None, 'El Monte', 'CA', 'False', 'other', 'Car', 'False'),
 (1533, 'Thongsoune Vilaysane', datetime.date(2016, 5, 11), 'shot', 'vehicle', 33, 'M', 'A', 'San Diego', 'CA', 'False', 'other', 'Car', 'True'),
 (1634, 'Sergey Makarenko', datetime.date(2016, 6, 18), 'shot', 'vehicle', 17, 'M', 'W', 'Carmichael', 'CA', 'False', 'other', 'Car', 'False'),
 (1691, 'Pedro Villanueva', datetime.date(2016, 7, 3), 'shot', 'vehicle', 19, 'M', 'H', 'Fullerton', 'CA', 'False', 'other', 'Car', 'False'),
 (1599, 'Omar Villagomez', datetime.date(2016, 6, 7), 'shot', 'vehicle', 21, 'M', 'H', 'Turlock', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (2006, 'Michelle Lee Shirley', datetime.date(2016, 10, 31), 'shot', 'vehicle', 39, 'F', 'B', 'Torrance', 'CA', 'True', 'attack', 'Car', 'False'),
 (1356, 'Gevork Alachadzhyan', datetime.date(2016, 3, 13), 'shot', 'vehicle', 45, 'M', 'W', 'Glendale', 'CA', 'True', 'attack', 'Car', 'False'),
 (1316, 'Francisco Garcia', datetime.date(2016, 2, 24), 'shot', 'vehicle', 26, 'M', 'H', 'Cerritos', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (1832, 'Eric Ortega Soto', datetime.date(2016, 8, 13), 'shot', 'vehicle', 36, 'M', 'H', 'Hayward', 'CA', 'False', 'other', 'Car', 'False'),
 (1162, 'Alan Paul Bartley', datetime.date(2016, 1, 11), 'shot', 'vehicle', 56, 'M', 'W', 'San Fernando', 'CA', 'False', 'attack', 'Car', 'False'),
 (1698, 'Adam Smith', datetime.date(2016, 7, 7), 'shot', 'vehicle', 33, 'M', 'W', 'Clovis', 'CA', 'False', 'other', 'Car', 'False'),
 (2053, 'Adalid Flores', datetime.date(2016, 11, 19), 'shot', 'vehicle', 29, 'M', 'H', 'Anaheim', 'CA', 'False', 'attack', 'Foot', 'True')]

No typo in the date

No typo in the city

Searching for an error within the same state, year and weapon: 

Google says Jose Sanchez, 30 latin

Killed by police says

In [47]:
%%sql

-- 13: 2164, from Gadsden,2016-12-24

select * from victims where city = 'Gadsden' order by name desc;

1 rows affected.


[(2164, None, datetime.date(2016, 12, 24), 'shot', 'gun', None, 'M', None, 'Gadsden', 'AL', 'False', 'attack', 'Foot', 'False')]

In [48]:
%sql select * from victims where date = '2016-12-24' order by name desc;

4 rows affected.


[(2164, None, datetime.date(2016, 12, 24), 'shot', 'gun', None, 'M', None, 'Gadsden', 'AL', 'False', 'attack', 'Foot', 'False'),
 (2166, 'Zhonghua Li', datetime.date(2016, 12, 24), 'shot', 'scissors', 48, 'M', 'A', 'Pico Rivera', 'CA', 'True', 'attack', 'Not fleeing', 'False'),
 (2163, 'Miriam Ann Savino', datetime.date(2016, 12, 24), 'shot', 'knife', 53, 'F', 'W', 'Lake Wales', 'FL', 'False', 'other', 'Not fleeing', 'False'),
 (2158, 'Daquan Antonio Westbrook', datetime.date(2016, 12, 24), 'shot', 'gun', 18, 'M', 'B', 'Charlotte', 'NC', 'False', 'undetermined', 'Not fleeing', 'False')]

In [ ]:
-- No typo in the date
-- No typo in the city
-- Google says Orande Kandie Hayes
-- Killed by police says

In [49]:
%%sql

-- 14: 2182,from Pensacola,2016-12-30

select * from victims where city = 'Pensacola' order by name desc;

1 rows affected.


[(2182, None, datetime.date(2016, 12, 30), 'shot', 'gun', None, 'M', None, 'Pensacola', 'FL', 'False', 'attack', 'Not fleeing', 'False')]

In [50]:
%sql select * from victims where date = '2016-12-30' order by name desc;

3 rows affected.


[(2182, None, datetime.date(2016, 12, 30), 'shot', 'gun', None, 'M', None, 'Pensacola', 'FL', 'False', 'attack', 'Not fleeing', 'False'),
 (2176, 'Jamal Rollins', datetime.date(2016, 12, 30), 'shot', 'gun', 21, 'M', 'B', 'Miami', 'FL', 'False', 'other', 'Car', 'False'),
 (2186, 'Dustin Selby', datetime.date(2016, 12, 30), 'shot and Tasered', 'knife', 31, 'M', None, 'Phoenix', 'AZ', 'False', 'other', 'Foot', 'False')]

In [ ]:
-- No typo in the date
-- Possible confussion with Jamal Rollins
-- Google says
-- Killed by police says Daniel Ralph Daily, 34 , white

In [51]:
%%sql

-- 15: 2232, from Los Angeles,2017-01-18

select * from victims where city = 'Los Angeles' order by name desc;

31 rows affected.


[(2232, None, datetime.date(2017, 1, 18), 'shot', 'hatchet', None, 'M', None, 'Los Angeles', 'CA', 'False', 'other', 'Car', 'False'),
 (2133, 'Ryan Joseph', datetime.date(2016, 12, 18), 'shot', 'undetermined', 20, 'M', 'B', 'Los Angeles', 'CA', 'False', 'undetermined', 'Foot', 'False'),
 (1105, 'Ruben Jose Herrera', datetime.date(2015, 12, 19), 'shot', 'undetermined', 26, 'M', 'H', 'Los Angeles', 'CA', 'True', 'attack', 'Not fleeing', 'False'),
 (331, 'Roberto Rodriguez', datetime.date(2015, 4, 8), 'shot', 'gun', 39, 'M', 'H', 'Los Angeles', 'CA', 'False', 'attack', 'Foot', 'False'),
 (1732, 'Richard Risher', datetime.date(2016, 7, 25), 'shot', 'gun', 18, 'M', 'B', 'Los Angeles', 'CA', 'False', 'attack', 'Foot', 'False'),
 (732, 'Redel Jones', datetime.date(2015, 8, 12), 'shot and Tasered', 'knife', 30, 'F', 'B', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (2085, 'Pablo Renato Cartagena', datetime.date(2016, 11, 28), 'shot', 'gun', 23, 'M', 'H', 'Los Angeles', 'CA', 'False', 'undetermined', 'Foot', 'False'),
 (75, 'Pablo Meza', datetime.date(2015, 1, 17), 'shot', 'gun', 24, 'M', 'H', 'Los Angeles', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (871, 'Norma Angelica Guzman', datetime.date(2015, 9, 27), 'shot', 'knife', 37, 'F', 'H', 'Los Angeles', 'CA', 'True', 'other', 'Not fleeing', 'True'),
 (627, 'Neil White', datetime.date(2015, 7, 9), 'shot and Tasered', 'Taser', 38, 'M', 'W', 'Los Angeles', 'CA', 'False', 'attack', 'Other', 'False'),
 (1001, 'Miguel Cano', datetime.date(2015, 11, 9), 'shot and Tasered', 'bean-bag gun', 35, 'M', 'H', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (2207, 'Michael Rogers', datetime.date(2017, 1, 10), 'shot', 'knife', 32, 'M', 'W', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (1508, 'Marion Habana', datetime.date(2016, 4, 30), 'shot', 'knife', 28, 'M', 'A', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'True'),
 (784, 'Manuel Soriano', datetime.date(2015, 8, 27), 'shot', "contractor's level", 29, 'M', 'H', 'Los Angeles', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (495, 'Luis Martinez', datetime.date(2015, 4, 21), 'shot', 'knife', 35, 'M', 'H', 'Los Angeles', 'CA', 'True', 'other', 'Not fleeing', 'False'),
 (1795, 'Kenney Watkins', datetime.date(2016, 8, 16), 'shot', 'gun', 18, 'M', 'B', 'Los Angeles', 'CA', 'False', 'other', 'Foot', 'False'),
 (1838, 'Keith Bursey', datetime.date(2016, 6, 10), 'shot', 'gun', 31, 'M', 'B', 'Los Angeles', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (1240, 'Jose Mendez', datetime.date(2016, 2, 6), 'shot', 'gun', 16, 'M', 'H', 'Los Angeles', 'CA', 'False', 'attack', 'Car', 'True'),
 (1776, 'Jesse James Romero', datetime.date(2016, 8, 9), 'shot', 'gun', 14, 'M', 'H', 'Los Angeles', 'CA', 'False', 'attack', 'Foot', 'True'),
 (619, 'Jason Hendley', datetime.date(2015, 7, 6), 'shot', 'knife', 29, 'M', 'B', 'Los Angeles', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (645, 'Jason Davis', datetime.date(2015, 7, 13), 'shot and Tasered', 'box cutter', 41, 'M', 'W', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (2134, 'Fidel Barrios', datetime.date(2016, 12, 17), 'shot', 'gun', 33, 'M', 'H', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (1271, 'Eduardo Rodriguez', datetime.date(2016, 2, 14), 'shot', 'gun', 29, 'M', 'H', 'Los Angeles', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (934, 'Dion Lamont Ramirez', datetime.date(2015, 10, 19), 'shot', 'vehicle', 53, 'M', 'B', 'Los Angeles', 'CA', 'False', 'attack', 'Not fleeing', 'False'),
 (1922, 'Daniel Perez', datetime.date(2016, 10, 2), 'shot', 'toy weapon', 16, 'M', 'H', 'Los Angeles', 'CA', 'True', 'attack', 'Not fleeing', 'True'),
 (194, 'Charly Leundeu Keunang', datetime.date(2015, 3, 1), 'shot and Tasered', 'unarmed', 43, 'M', 'B', 'Los Angeles', 'CA', 'True', 'attack', 'Not fleeing', 'True'),
 (1925, 'Carnell Snell', datetime.date(2016, 10, 1), 'shot', 'gun', 18, 'M', 'B', 'Los Angeles'

In [52]:
%sql select * from victims where date = '2017-01-18' order by name desc;

7 rows affected.


[(2232, None, datetime.date(2017, 1, 18), 'shot', 'hatchet', None, 'M', None, 'Los Angeles', 'CA', 'False', 'other', 'Car', 'False'),
 (2230, 'Troy Bateman', datetime.date(2017, 1, 18), 'shot', 'gun', 35, 'M', 'W', 'Columbia', 'MO', 'False', 'attack', 'Not fleeing', 'False'),
 (2228, 'Stoney McJunkin', datetime.date(2017, 1, 18), 'shot', 'metal pipe', 40, 'M', 'W', 'Watts', 'OK', 'False', 'attack', 'Not fleeing', 'False'),
 (2245, 'Melvin DeLong', datetime.date(2017, 1, 18), 'shot', 'gun', 28, 'M', 'N', 'Grand Forks', 'ND', 'False', 'attack', 'Not fleeing', 'False'),
 (2243, 'Marquis Thomas', datetime.date(2017, 1, 18), 'shot', 'gun', 19, 'M', 'B', 'Gary', 'IN', 'False', 'attack', 'Car', 'False'),
 (2225, 'Marlon Joel Rodas-Sanchez', datetime.date(2017, 1, 18), 'shot and Tasered', 'knife', 16, 'M', 'H', 'Salinas', 'CA', 'False', 'other', 'Not fleeing', 'True'),
 (2231, 'Genevive Dawes', datetime.date(2017, 1, 18), 'shot', 'vehicle', 21, 'F', 'W', 'Dallas', 'TX', 'False', 'other', 'Car', 'False')]

In [53]:
%sql select * from victims where armed = 'hatchet' order by name desc;

6 rows affected.


[(2232, None, datetime.date(2017, 1, 18), 'shot', 'hatchet', None, 'M', None, 'Los Angeles', 'CA', 'False', 'other', 'Car', 'False'),
 (707, 'Vincente Montano', datetime.date(2015, 8, 5), 'shot', 'hatchet', 29, 'M', 'W', 'Antioch', 'TN', 'True', 'attack', 'Not fleeing', 'False'),
 (1890, 'Robert Duh', datetime.date(2016, 9, 17), 'shot', 'hatchet', 42, 'M', 'W', 'Beaumont', 'CA', 'False', 'other', 'Not fleeing', 'False'),
 (1446, 'Melissa M. Abbott', datetime.date(2016, 4, 8), 'shot', 'hatchet', 25, 'F', 'W', 'Lake Hallie', 'WI', 'True', 'other', 'Not fleeing', 'False'),
 (153, 'Bruce Steward', datetime.date(2015, 2, 15), 'shot', 'hatchet', 34, 'M', 'W', 'Colton', 'OR', 'True', 'other', 'Not fleeing', 'False'),
 (1707, 'Benjamin Zeckovic', datetime.date(2016, 7, 12), 'shot', 'hatchet', 23, 'M', 'W', 'Anchorage', 'AK', 'True', 'attack', 'Not fleeing', 'False')]

In [ ]:
-- Impossible to fins a typo in the date...a lot of entries in LA
-- No typo in the city
-- Strange guy fleeing by car with an hatchet...no light neither
-- Google says
-- Killed by police says David Darnell Stroughter, 50 , Black, Westchester (neighbourhood of LA?)

In [54]:
%%sql

-- 16: 2278 ,from Hollywood,2017-01-31

select * from victims where city = 'Hollywood' order by name desc;

1 rows affected.


[(2278, None, datetime.date(2017, 1, 31), 'shot and Tasered', 'knife', None, 'M', 'B', 'Hollywood', 'CA', 'True', 'attack', 'Not fleeing', 'False')]

In [55]:
%sql select * from victims where date = '2017-01-31' order by name desc;

1 rows affected.


[(2278, None, datetime.date(2017, 1, 31), 'shot and Tasered', 'knife', None, 'M', 'B', 'Hollywood', 'CA', 'True', 'attack', 'Not fleeing', 'False')]

In [ ]:
-- No typo in the date
-- No typo in the city
-- Google says
-- Killed by police says Solomon Picart, 37, Black

In [56]:
%%sql

-- 17: 2319,from Crownpoint,2017-02-09

select * from victims where city = 'Crownpoint' order by name desc;

1 rows affected.


[(2319, None, datetime.date(2017, 2, 9), 'shot', 'undetermined', None, 'M', None, 'Crownpoint', 'NM', 'False', 'undetermined', None, 'False')]

In [57]:
%sql select * from victims where date = '2017-02-09' order by name desc;

4 rows affected.


[(2319, None, datetime.date(2017, 2, 9), 'shot', 'undetermined', None, 'M', None, 'Crownpoint', 'NM', 'False', 'undetermined', None, 'False'),
 (2313, 'Vincent Palma', datetime.date(2017, 2, 9), 'shot and Tasered', 'unarmed', 26, 'M', 'W', 'Geneva Township', 'OH', 'False', 'undetermined', 'Not fleeing', 'False'),
 (2316, 'Quanice Derrick Hayes', datetime.date(2017, 2, 9), 'shot', 'toy weapon', 17, 'M', 'B', 'Portland', 'OR', 'False', 'undetermined', 'Foot', 'False'),
 (2315, 'Juan Salinas', datetime.date(2017, 2, 9), 'shot and Tasered', 'knife', 33, 'M', 'H', 'Lake Stevens', 'WA', 'True', 'other', 'Not fleeing', 'False')]

In [ ]:
-- No typo in the date
-- No typo in the city
-- Google says nothing
-- Killed by police says nothing

In [58]:
%%sql

-- 18: 2328, from Terrell,2017-02-12

select * from victims where city = 'Terrell' order by name desc;

1 rows affected.


[(2328, None, datetime.date(2017, 2, 12), 'shot', 'vehicle', None, 'M', None, 'Terrell', 'TX', 'False', 'other', 'Other', 'False')]

In [59]:
%sql select * from victims where date = '2017-02-12' order by name desc;

3 rows affected.


[(2328, None, datetime.date(2017, 2, 12), 'shot', 'vehicle', None, 'M', None, 'Terrell', 'TX', 'False', 'other', 'Other', 'False'),
 (2329, 'Willard Eugene Scott', datetime.date(2017, 2, 12), 'shot', 'gun', 31, 'M', 'B', 'Durham', 'NC', 'False', 'undetermined', 'Other', 'False'),
 (2333, 'Brian Carreno', datetime.date(2017, 2, 12), 'shot', 'knife', 26, 'M', None, 'Santa Barbara', 'CA', 'True', 'other', 'Foot', 'False')]

In [ ]:
-- No typo in the date
-- No typo in the city
-- Google says nothing
-- Killed by police says nothing

In [61]:
%%sql 

-- 19: 2384, from Lumpkin,2017-02-25

select * from victims where city = 'Lumpkin' order by name desc;

1 rows affected.


[(2384, None, datetime.date(2017, 2, 25), 'shot', 'gun', None, None, None, 'Lumpkin', 'GA', 'False', 'attack', 'Car', 'False')]

In [62]:
%sql select * from victims where date = '2017-02-25' order by name desc;

2 rows affected.


[(2384, None, datetime.date(2017, 2, 25), 'shot', 'gun', None, None, None, 'Lumpkin', 'GA', 'False', 'attack', 'Car', 'False'),
 (2383, 'Lorenzo Antoine Cruz', datetime.date(2017, 2, 25), 'shot', 'toy weapon', 23, 'M', None, 'Rocklin', 'CA', 'False', 'attack', 'Foot', 'False')]

* No typo in the date
* No typo in the city
* Google says William Dwayne Darby, 39, White
* Killed by police says